ac_new.rb

```diff
@@ -1,388 +1,367 @@
     # read mandatory ENV params
     # I love Ruby but I find instance_variable_set() and instance_variable_get() messy and not worth it...
-    # this is actually one of these mighty rare occasions where PHP outshines ruby in terms of cleanliness of syntax with the `$$` or variable variable. 
+    # this is actually one of these mighty rare occasions where PHP outshines ruby in terms of cleanliness of syntax with the `$$` or variable variable.
     # so, verify existence first and set later.. less "cool" but you can't have it all:)
-    
-    mandatory_env_vars = ['OUTDIR', 'MEETING_ID', 'MEETING_NAME', 'CATEGORY_NAME', 'AC_USERNAME', 'AC_PASSWD', 'X_SERVER_DISPLAY_NUM'] 
-    
+
+    mandatory_env_vars = %w[OUTDIR MEETING_ID MEETING_NAME CATEGORY_NAME AC_USERNAME AC_PASSWD X_SERVER_DISPLAY_NUM]
+
     mandatory_env_vars.each do |mandatory_var|
-      if ! ENV[mandatory_var] or ENV[mandatory_var].empty?
-	@logger.error('Missing ENV var ' + mandatory_var + ':( Make sure you export it.')
-	return false
+      if !ENV[mandatory_var] || ENV[mandatory_var].empty?
+        @logger.error('Missing ENV var ' + mandatory_var + ':( Make sure you export it.')
+        return false
       end
     end
 
-    out_dir=ENV['OUTDIR'] + '/'
+    out_dir = ENV['OUTDIR'] + '/'
     # let's strip spaces just to be on the safe side
-    meeting_id=ENV['MEETING_ID'].strip
-    entry_name=ENV['MEETING_NAME']
-    cat_name=ENV['CATEGORY_NAME']
-    ac_username=ENV['AC_USERNAME']
-    ac_passwd=ENV['AC_PASSWD']
-    x_display=ENV['X_SERVER_DISPLAY_NUM']
-
-    if ENV['FFMPEG_BIN']
-	    ffmpeg_bin=ENV['FFMPEG_BIN']
-    else
-	    ffmpeg_bin='ffmpeg'
-    end
+    meeting_id = ENV['MEETING_ID'].strip
+    entry_name = ENV['MEETING_NAME']
+    cat_name = ENV['CATEGORY_NAME']
+    ac_username = ENV['AC_USERNAME']
+    ac_passwd = ENV['AC_PASSWD']
+    x_display = ENV['X_SERVER_DISPLAY_NUM']
 
-    if ENV['FFPROBE_BIN']
-	    ffprobe_bin=ENV['FFPROBE_BIN']
-    else
-	    ffprobe_bin='ffprobe'
-    end
-    
-    resolution='1280x720'
-    frame_rate='30'
-    audio_file=out_dir + meeting_id + '.mp3'
-    recording_file=out_dir + meeting_id + '.mkv'
-    full_recording_file=out_dir + meeting_id + '.full.mkv'
-	
-    basedir=File.dirname(__FILE__) 
+    ffmpeg_bin = ENV['FFMPEG_BIN'] || 'ffmpeg'
+
+    ffprobe_bin = ENV['FFPROBE_BIN'] || 'ffprobe'
+
+    resolution = '1280x720'
+    frame_rate = '30'
+    audio_file = out_dir + meeting_id + '.mp3'
+    recording_file = out_dir + meeting_id + '.mkv'
+    full_recording_file = out_dir + meeting_id + '.full.mkv'
+
+    basedir = File.dirname(__FILE__)
     @logger.info(basedir + '/get_ac_audio.sh ' + meeting_id)
-    system basedir + '/get_ac_audio.sh ' + meeting_id 
-    if $?.exitstatus != 0 or ! File.exist?(audio_file)
-	    @logger.error("Failed to obtain audio file :(")
-	    return false
+    system basedir + '/get_ac_audio.sh ' + meeting_id
+    if ($CHILD_STATUS.exitstatus != 0) || !File.exist?(audio_file)
```

Comment:
Sorry, but I much prefer `$?` over `$CHILD_STATUS`. Also, in order to use $CHILD_STATUS, one must require the `English` module. Consider this snippet:
```ruby
#require 'English'

system('pwd')

fail unless $?.success?
fail unless $CHILD_STATUS.success?
```
Which will result in: 
> `<main>': undefined method `success?' for nil:NilClass (NoMethodError)
unless you uncomment:
`#require 'English'`